homeassistant/helpers/collection.py

```diff
@@ -326,20 +359,22 @@ def sync_entity_lifecycle(
     domain: str,
     platform: str,
     entity_component: EntityComponent,
-    collection: ObservableCollection,
-    create_entity: Callable[[dict], Entity],
+    collection: StorageCollection | YamlCollection,
+    entity_class: type[CollectionEntity],
```

Comment:
Do we need a dev blog for this change?

Reply:
There are no custom components published on HACS which use our collection helper. We can still publish a blog post in case some non-HACS components is using the collections helper though.